backend/src/build-system/node/ux-sitemap-structure/index.ts

```diff
@@ -16,13 +16,22 @@ export class UXSitemapStructureHandler implements BuildHandler {
     const projectName =
       context.getData('projectName') || 'Default Project Name';
 
+    const sitemap = args[0] as string;
+
+    if (!sitemap) {
+      return {
+        success: false,
+        error: new Error('Missing required parameters: sitemap'),
```

Comment:
we should make global context error tracker, when some generation error happning, we should record it